test/src/sdk-harness/test_projects/generics_in_abi/src/main.sw

```diff
@@ -79,7 +79,7 @@ impl MyContract for Contract {
     }
 
     fn struct_w_generic_in_tuple(arg1: StructWTupleGeneric<u32>) -> StructWTupleGeneric<u32> {
-        let expected = StructWTupleGeneric { a: (1, 2) };
+        let expected = StructWTupleGeneric { a: (1u32, 2u32) };
```

Comment:
Same question as above: as a Sway user I would prefer to write `(1, 2)` instead of `(1u32, 2u32)` and rely on the compiler to infer the type of the numeric literals

Reply:
It won't work in this case because it's ambiguous.

The way our type checking works right now, the type of `expected` in `let expected = StructWTupleGeneric { a: (1, 2) };` has to be resolved to a concrete type, so it can't be `StructWTupleGeneric<numeric>` because that's not something that can be impl'd for. Hence we decay the numerics in type arguments to `u64`.

Had we a more powerful type checking implementation we could probably do better, but such changes are too large to be in scope for this I think.

Reply:
Got it, thanks. Probably would be a good idea to open an issue on things like this after the PR is merged